bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java

```diff
@@ -235,12 +244,18 @@ public ResultArrayIterator(int arraySize) {
     }
 
     @Override
-    public Result[] next(ResultScanner<Row> resultScanner)
+    public Result[] next(ResultScanner<Row> resultScanner, ByteKeyRangeTracker rangeTracker)
         throws IOException {
       List<Result> results = new ArrayList<>();
       for (int i = 0; i < arraySize; i++) {
         Row row = resultScanner.next();
         if (row == null) {
+          // The scan completed.
+          break;
+        }
+        ByteKey key = ByteKey.copyFrom(row.getKey().toByteArray());
```

Comment:
Not sure what to do about it but this seems to make 2 array copies for every row which isn't ideal.